common/client/node_fsm.go

```diff
@@ -10,136 +10,136 @@ import (
 var (
 	promPoolRPCNodeTransitionsToAlive = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "pool_rpc_node_num_transitions_to_alive",
-		Help: transitionString(nodeStateAlive),
+		Help: transitionString(NodeStateAlive),
 	}, []string{"chainID", "nodeName"})
 	promPoolRPCNodeTransitionsToInSync = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "pool_rpc_node_num_transitions_to_in_sync",
-		Help: fmt.Sprintf("%s to %s", transitionString(nodeStateOutOfSync), nodeStateAlive),
+		Help: fmt.Sprintf("%s to %s", transitionString(NodeStateOutOfSync), NodeStateAlive),
 	}, []string{"chainID", "nodeName"})
 	promPoolRPCNodeTransitionsToOutOfSync = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "pool_rpc_node_num_transitions_to_out_of_sync",
-		Help: transitionString(nodeStateOutOfSync),
+		Help: transitionString(NodeStateOutOfSync),
 	}, []string{"chainID", "nodeName"})
 	promPoolRPCNodeTransitionsToUnreachable = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "pool_rpc_node_num_transitions_to_unreachable",
-		Help: transitionString(nodeStateUnreachable),
+		Help: transitionString(NodeStateUnreachable),
 	}, []string{"chainID", "nodeName"})
 	promPoolRPCNodeTransitionsToInvalidChainID = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "pool_rpc_node_num_transitions_to_invalid_chain_id",
-		Help: transitionString(nodeStateInvalidChainID),
+		Help: transitionString(NodeStateInvalidChainID),
 	}, []string{"chainID", "nodeName"})
 	promPoolRPCNodeTransitionsToUnusable = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "pool_rpc_node_num_transitions_to_unusable",
-		Help: transitionString(nodeStateUnusable),
+		Help: transitionString(NodeStateUnusable),
 	}, []string{"chainID", "nodeName"})
 	promPoolRPCNodeTransitionsToSyncing = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "pool_rpc_node_num_transitions_to_syncing",
-		Help: transitionString(nodeStateSyncing),
+		Help: transitionString(NodeStateSyncing),
 	}, []string{"chainID", "nodeName"})
 )
 
-// nodeState represents the current state of the node
+// NodeState represents the current state of the node
 // Node is a FSM (finite state machine)
-type nodeState int
+type NodeState int
```

Comment:
Why does this need to be exported now?

Reply:
A map of node states is returned by MultiNode now and uses this type

Reply:
Is that in the "external" API? Or just used by each chain implementation for the "internal" API of multinode?

Reply:
It's used by the chain implementation outside of the common components which is why we decided to export the type for simplicity (previously we were copying all of the states to strings rather than using this type)

Reply:
Alright, changed back.